selfdrive/ui/qt/maps/map.cc

```diff
@@ -378,46 +383,32 @@ void MapWindow::updateDestinationMarker() {
   }
 }
 
-MapInstructions::MapInstructions(QWidget * parent) : QWidget(parent) {
+MapInstructions::MapInstructions(QWidget *parent) : QWidget(parent) {
   is_rhd = Params().getBool("IsRhdDetected");
   QHBoxLayout *main_layout = new QHBoxLayout(this);
   main_layout->setContentsMargins(11, 50, 11, 11);
-  {
-    QVBoxLayout *layout = new QVBoxLayout;
-    icon_01 = new QLabel;
-    layout->addWidget(icon_01);
-    layout->addStretch();
-    main_layout->addLayout(layout);
-  }
-
-  {
-    QVBoxLayout *layout = new QVBoxLayout;
-
-    distance = new QLabel;
-    distance->setStyleSheet(R"(font-size: 90px;)");
-    layout->addWidget(distance);
+  main_layout->addWidget(icon_01 = new QLabel, 0, Qt::AlignTop);
 
-    primary = new QLabel;
-    primary->setStyleSheet(R"(font-size: 60px;)");
-    primary->setWordWrap(true);
-    layout->addWidget(primary);
+  QWidget *right_container = new QWidget(this);
+  right_container->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Preferred);
+  QVBoxLayout *layout = new QVBoxLayout(right_container);
 
-    secondary = new QLabel;
-    secondary->setStyleSheet(R"(font-size: 50px;)");
-    secondary->setWordWrap(true);
-    layout->addWidget(secondary);
+  layout->addWidget(distance = new QLabel);
+  distance->setStyleSheet(R"(font-size: 90px;)");
 
-    lane_widget = new QWidget;
-    lane_widget->setFixedHeight(125);
+  layout->addWidget(primary = new QLabel);
+  primary->setStyleSheet(R"(font-size: 60px;)");
+  primary->setWordWrap(true);
 
-    lane_layout = new QHBoxLayout(lane_widget);
-    layout->addWidget(lane_widget);
+  layout->addWidget(secondary = new QLabel);
+  secondary->setStyleSheet(R"(font-size: 50px;)");
+  secondary->setWordWrap(true);
 
-    main_layout->addLayout(layout);
-  }
+  layout->addLayout(lane_layout = new QHBoxLayout);
+  layout->addStretch();
```

Comment:
@incognitojam Not sure if we need this stretch, if there is only one message it will make the message always be displayed at the top. 
The  original version  didn't have such a stretch. it will centers labels vertically in the available space.
what do you think ? should we keep this stretch or remove it?

Reply:
Should be fine without